File: src/main/java/me/kuku/yuq/entity/BiliBiliEntity.java
package me.kuku.yuq.entity;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONArray;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "bilibili")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class BiliBiliEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true)
    private Long qq;
    @Column(name = "group_")
    private Long group;
    @Lob
    @Column(columnDefinition = "text")
    private String cookie;
    private Boolean monitor;
    @Column(name = "task_")
    private Boolean task;
    @Lob
    @Column(columnDefinition = "text")
    private String liveList;
    @Lob
    @Column(columnDefinition = "text")
    private String likeList;
    @Lob
    @Column(columnDefinition = "text")
    private String commentList;
    @Lob
    @Column(columnDefinition = "text")
    private String forwardList;
    @Lob
    @Column(columnDefinition = "text")
    private String tossCoinList;
    @Lob
    @Column(columnDefinition = "text")
    private String favoritesList;
    private String token;
    private String userId;

    public BiliBiliEntity(String cookie, String userId, String token){
        this.cookie = cookie;
        this.userId = userId;
        this.token = token;
    }

    public BiliBiliEntity(Long qq, Long group){
        this.qq = qq;
        this.group = group;
    }

    public void setLiveJsonArray(JSONArray jsonArray){
        this.liveList = jsonArray.toString();
    }

    public JSONArray getLiveJsonArray(){
        if (liveList == null) return JSON.parseArray("[]");
        return JSON.parseArray(liveList);
    }

    public void setLikeJsonArray(JSONArray jsonArray){
        this.likeList = jsonArray.toString();
    }

    public JSONArray getLikeJsonArray(){
        if (likeList == null) return JSON.parseArray("[]");
        return JSON.parseArray(likeList);
    }

    public void setCommentJsonArray(JSONArray jsonArray){
        this.commentList = jsonArray.toString();
    }

    public JSONArray getCommentJsonArray(){
        if (commentList == null) return JSON.parseArray("[]");
        return JSON.parseArray(commentList);
    }

    public void setForwardJsonArray(JSONArray jsonArray){
        this.forwardList = jsonArray.toString();
    }

    public JSONArray getForwardJsonArray(){
        if (forwardList == null) return JSON.parseArray("[]");
        return JSON.parseArray(forwardList);
    }

    public void setTossCoinJsonArray(JSONArray jsonArray){
        this.tossCoinList = jsonArray.toString();
    }

    public JSONArray getTossCoinJsonArray(){
        if (tossCoinList == null) return JSON.parseArray("[]");
        return JSON.parseArray(tossCoinList);
    }

    public void setFavoritesJsonArray(JSONArray jsonArray){
        this.favoritesList = jsonArray.toString();
    }

    public JSONArray getFavoritesJsonArray(){
        if (favoritesList == null) return JSON.parseArray("[]");
        return JSON.parseArray(favoritesList);
    }
}

File: src/main/kotlin/me/kuku/yuq/entity/BiliBiliEntity.kt
package me.kuku.yuq.entity

import org.hibernate.annotations.Type
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import javax.persistence.*

@Entity
@Table(name = "bili_bili")
@NamedEntityGraph(name = "bili_bili_qq", attributeNodes = [NamedAttributeNode(value = "qqEntity", subgraph = "qqEntity")],
    subgraphs = [NamedSubgraph(name = "qqEntity", attributeNodes = [NamedAttributeNode("groups")])])
class BiliBiliEntity: BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
    @OneToOne
    @JoinColumn(name = "qq_id")
    var qqEntity: QqEntity? = null
    @Column(length = 2000)
    var cookie: String = ""
    var userid: String = ""
    var token: String = ""
    @Type(type = "json")
    @Column(columnDefinition = "json")
    var config: BiliBiliConfig = BiliBiliConfig()
}

interface BiliBiliRepository: JpaRepository<BiliBiliEntity, Int> {
    fun findByQqEntity(qqEntity: QqEntity): BiliBiliEntity?

    @EntityGraph(value = "bili_bili_qq", type = EntityGraph.EntityGraphType.FETCH)
    override fun findAll(): List<BiliBiliEntity>
}

@Service
class BiliBiliService (
    private val biliBiliRepository: BiliBiliRepository
) {
    fun findByQqEntity(qqEntity: QqEntity) = biliBiliRepository.findByQqEntity(qqEntity)

    fun save(biliEntity: BiliBiliEntity): BiliBiliEntity = biliBiliRepository.save(biliEntity)

    fun findAll(): List<BiliBiliEntity> = biliBiliRepository.findAll()

}

data class BiliBiliConfig(var push: Status = Status.OFF, var sign: Status = Status.OFF, var live: Status = Status.OFF, var coin: Status = Status.OFF)